### PR TITLE
Allow overwriting host and timestamp

### PR DESCRIPTION
**Pull request by indrekj**

### CODE REVIEW

File: spec/utils_spec.rb [hunk 1]
```diff
@@ -33,5 +33,23 @@
...
+      it 'does not overwrite @timestamp' do
+        expect(subject[:message]).to eq('test')
+        expect(subject[:@timestamp]).to eq('a timestamp')
+      end
+    end
+
+    context 'when host provided' do
+      let(:metadata) { {message: 'test', host: 'xyz'} }
+
+      it 'does not overwrite @timestamp' do
```
Comment (deiwin) [inline spec/utils_spec.rb]: `@timestamp` -> `host`

Comment (deiwin) [inline spec/utils_spec.rb]: Also, this and the above `it` block say 'doesn't overwrite', but you call them 'overwritable params' in the code. Which is it?

File: lib/logasm/utils.rb [hunk 2]
```diff
@@ -12,12 +12,12 @@ module Utils
     #
     # @return [Hash]
     def self.build_event(metadata, level, service_name)
-      metadata.merge(
-        application: application_name(service_name),
-        level: level.to_s.downcase,
-        host: HOST,
-        :@timestamp => Time.now.utc.iso8601(3)
-      )
+      overwrittable_params
```
Comment (deiwin) [inline lib/logasm/utils.rb]: single 't'

**Comment by indrekj** [comment]
updated

File: lib/logasm/utils.rb [hunk 3]
```diff
@@ -31,5 +31,13 @@ def self.build_event(metadata, level, service_name)
     def self.application_name(service_name)
       Inflecto.underscore(service_name)
     end
+
+    def self.overwritable_params
+      {
+        :@timestamp => Time.now.utc.iso8601(3),
```
Comment (deiwin) [inline lib/logasm/utils.rb]: what's that `3` doing here?

**Comment by deiwin** [comment]
Why does the `timestamp` need to be `@` prefixed?

File: lib/logasm/utils.rb (hunk 3 above)
Comment (deiwin) [inline lib/logasm/utils.rb]: Does that mean, that the default value is 3 units from the epoch?

Comment (deiwin) [inline lib/logasm/utils.rb]: Although, the `now` wouldn't make sense then

Comment (indrekj) [inline lib/logasm/utils.rb]: yep, 3 units. "..T17:12:00.XXXZ"

Comment (deiwin) [inline lib/logasm/utils.rb]: I would like to see this number unmagicked

Comment (indrekj) [inline lib/logasm/utils.rb]: @SteamRiC any reason this has to be 3? why doesn't 0 or 5 work?

Comment (SteamRiC) [inline lib/logasm/utils.rb]: If I remember correctly, our old logging had this timestamp format and 3 was just to make it similar to that.

**Comment by indrekj** [comment]
Because of logstash

**Comment by deiwin** [comment]
+1

**Comment by SteamRiC** [comment]
!squash

**Pull request merged**